src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultClaimUidExtractor.cs

```diff
@@ -38,6 +38,17 @@ public string ExtractClaimUid(ClaimsIdentity claimsIdentity)
         // Internal for testing
         internal static IEnumerable<string> GetUniqueIdentifierParameters(ClaimsIdentity claimsIdentity)
         {
+            var subClaim = claimsIdentity.FindFirst(
+                claim => string.Equals("sub", claim.Type, StringComparison.Ordinal));
+            if (subClaim != null && !string.IsNullOrEmpty(subClaim.Value))
+            {
+                return new string[]
+                {
+                    "sub",
+                    subClaim.Value
```

Comment:
You'd take the claim value, and the claim issuer (the Issuer property on the claim), and safely concatenate them (basically you do len(value)+value+len(issuer)+issuer), then either put that in the token, or a SHA256 hash of the concatenated value.

This basically means that if, for example, facebook and google issued the same sub claim value, you could still tell them apart, because the issuer is different.